src/mongocxx/collection.cpp

```diff
@@ -1149,10 +1149,13 @@ cursor collection::_distinct(const client_session* session,
         throw bsoncxx::exception{bsoncxx::error_code::k_internal_error};
     }
 
+    scoped_bson_t error_document;
+    const bson_t *error_doc = error_document.bson_for_init();
```

Comment:
`scoped_bson_t` is useful for cases where we are initializing a `bson_t` and need to destroy it on scope exit (similar to why defer statement in Go), since it destroys in the destructor.

But I believe in this case, where `mongoc_cursor_error_document` is giving us a pointer to a `const bson_t*`, we don't want to call `bson_destroy` on the `bson_t` it gives us. The cursor manages the lifetime of that `bson_t`. The `const bson_t *` is a non-owning view of it.

I suggest removing `scoped_bson_t` and simplifying this to:

```
const bson_t *error_doc;
```

Reply:
Gotcha; switched back to just `const bson_t*` since the cursor is managing the lifetime of the error document.